monai/bundle/workflows.py

```diff
@@ -226,6 +246,114 @@ def check_properties(self) -> list[str] | None:
         return [n for n, p in self.properties.items() if p.get(BundleProperty.REQUIRED, False) and not hasattr(self, n)]
 
 
+class PythonicWorkflow(BundleWorkflow):
+    """
+    Base class for the pythonic workflow specification in bundle, it can be a training, evaluation or inference workflow.
+    It defines the basic interfaces for the bundle workflow behavior: `initialize`, `run`, `finalize`, etc.
+    This also provides the interface to get / set public properties to interact with a bundle workflow through
+    defined `get_<property>` accessor methods or directly defining members of the object.
```

Comment:
Please also describe how to `set` properties.
And better to talk about the lifecycle of properties with some examples: set, initialize, get, re-set, initialize, add property, etc.
Add property: https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/workflows.py#L200
Refer to this example for all the usage cases:
https://github.com/Project-MONAI/tutorials/blob/main/bundle/python_bundle_workflow/scripts/train.py#L84

Thanks.

Reply:
Add more description in the docstring and in the test case, please take a look at it, thanks.